[Bug] Can't try docker install on local on Macbook with M1 chip

## Description

Cannot try appsmith on local machine with docker

### Steps to reproduce the behaviour:
Add steps to reproduce this behaviour, include console / network logs & screenshots
1. Download the install.sh file
2. run it using `sh ./install.sh`
3. select the required options
4. containers start and then the internal server container dies.

### Important Details
 - Version: Self-Hosted latest
 - OS: MacOSX
 - Browser edge

Error Log:
```
appsmith-internal-server_1  | SLF4J: Class path contains multiple SLF4J bindings.
appsmith-internal-server_1  | SLF4J: Found binding in [jar:file:/server.jar!/BOOT-INF/lib/logback-classic-1.2.3.jar!/org/slf4j/impl/StaticLoggerBinder.class]
appsmith-internal-server_1  | SLF4J: Found binding in [jar:file:/server.jar!/BOOT-INF/lib/slf4j-log4j12-1.7.30.jar!/org/slf4j/impl/StaticLoggerBinder.class]
appsmith-internal-server_1  | SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
appsmith-internal-server_1  | SLF4J: Actual binding is of type [ch.qos.logback.classic.util.ContextSelectorStaticBinder]
nginx_1                     | 2021/05/22 12:21:26 [emerg] 59#59: io_setup() failed (38: Function not implemented)
appsmith-internal-server_1  | 
appsmith-internal-server_1  |   .   ____          _            __ _ _
appsmith-internal-server_1  |  /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
appsmith-internal-server_1  | ( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
appsmith-internal-server_1  |  \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
appsmith-internal-server_1  |   '  |____| .__|_| |_|_| |_\__, | / / / /
appsmith-internal-server_1  |  =========|_|==============|___/=/_/_/_/
appsmith-internal-server_1  |  :: Spring Boot ::        (v2.2.4.RELEASE)
appsmith-internal-server_1  | 
appsmith-internal-server_1  |  - Starting ServerApplication vv1.5.2 on dacd7db5d8e7 with PID 9 (/server.jar started by root in /)
appsmith-internal-server_1  |  - Running with Spring Boot v2.2.4.RELEASE, Spring v5.2.3.RELEASE
appsmith-internal-server_1  |  - No active profile set, falling back to default profiles: default
appsmith-internal-server_1  |  - Multiple Spring Data modules found, entering strict repository configuration mode!
appsmith-internal-server_1  |  - Bootstrapping Spring Data Reactive MongoDB repositories in DEFAULT mode.
appsmith-internal-server_1  |  - Finished Spring Data repository scanning in 1247ms. Found 22 Reactive MongoDB repository interfaces.
appsmith-internal-server_1  |  - Multiple Spring Data modules found, entering strict repository configuration mode!
appsmith-internal-server_1  |  - Bootstrapping Spring Data Redis repositories in DEFAULT mode.
appsmith-internal-server_1  |  - Spring Data Redis - Could not safely identify store assignment for repository candidate interface com.appsmith.server.repositories.PasswordResetTokenRepository. If you want this repository to be a Redis repository, consider annotating your entities with one of these annotations: org.springframework.data.redis.core.RedisHash (preferred), or consider extending one of the following types with your repository: org.springframework.data.keyvalue.repository.KeyValueRepository.
appsmith-internal-server_1  |  - Spring Data Redis - Could not safely identify store assignment for repository candidate interface com.appsmith.server.repositories.CommentThreadRepository. If you want this repository to be a Redis repository, consider annotating your entities with one of these annotations: org.springframework.data.redis.core.RedisHash (preferred), or consider extending one of the following types with your repository: org.springframework.data.keyvalue.repository.KeyValueRepository.
appsmith-internal-server_1  |  - Spring Data Redis - Could not safely identify store assignment for repository candidate interface com.appsmith.server.repositories.PageRepository. If you want this repository to be a Redis repository, consider annotating your entities with one of these annotations: org.springframework.data.redis.core.RedisHash (preferred), or consider extending one of the following types with your repository: org.springframework.data.keyvalue.repository.KeyValueRepository.
appsmith-internal-server_1  |  - Spring Data Redis - Could not safely identify store assignment for repository candidate interface com.appsmith.server.repositories.NewActionRepository. If you want this repository to be a Redis repository, consider annotating your entities with one of these annotations: org.springframework.data.redis.core.RedisHash (preferred), or consider extending one of the following types with your repository: org.springframework.data.keyvalue.repository.KeyValueRepository.
appsmith-internal-server_1  |  - Spring Data Redis - Could not safely identify store assignment for repository candidate interface com.appsmith.server.repositories.UserDataRepository. If you want this repository to be a Redis repository, consider annotating your entities with one of these annotations: org.springframework.data.redis.core.RedisHash (preferred), or consider extending one of the following types with your repository: org.springframework.data.keyvalue.repository.KeyValueRepository.
appsmith-internal-server_1  |  - Spring Data Redis - Could not safely identify store assignment for repository candidate interface com.appsmith.server.repositories.DatasourceRepository. If you want this repository to be a Redis repository, consider annotating your entities with one of these annotations: org.springframework.data.redis.core.RedisHash (preferred), or consider extending one of the following types with your repository: org.springframework.data.keyvalue.repository.KeyValueRepository.
appsmith-internal-server_1  |  - Spring Data Redis - Could not safely identify store assignment for repository candidate interface com.appsmith.server.repositories.PermissionRepository. If you want this repository to be a Redis repository, consider annotating your entities with one of these annotations: org.springframework.data.redis.core.RedisHash (preferred), or consider extending one of the following types with your repository: org.springframework.data.keyvalue.repository.KeyValueRepository.
appsmith-internal-server_1  |  - Spring Data Redis - Could not safely identify store assignment for repository candidate interface com.appsmith.server.repositories.ActionRepository. If you want this repository to be a Redis repository, consider annotating your entities with one of these annotations: org.springframework.data.redis.core.RedisHash (preferred), or consider extending one of the following types with your repository: org.springframework.data.keyvalue.repository.KeyValueRepository.
appsmith-internal-server_1  |  - Spring Data Redis - Could not safely identify store assignment for repository candidate interface com.appsmith.server.repositories.CommentRepository. If you want this repository to be a Redis repository, consider annotating your entities with one of these annotations: org.springframework.data.redis.core.RedisHash (preferred), or consider extending one of the following types with your repository: org.springframework.data.keyvalue.repository.KeyValueRepository.
appsmith-internal-server_1  |  - Spring Data Redis - Could not safely identify store assignment for repository candidate interface com.appsmith.server.repositories.ConfigRepository. If you want this repository to be a Redis repository, consider annotating your entities with one of these annotations: org.springframework.data.redis.core.RedisHash (preferred), or consider extending one of the following types with your repository: org.springframework.data.keyvalue.repository.KeyValueRepository.
appsmith-internal-server_1  |  - Spring Data Redis - Could not safely identify store assignment for repository candidate interface com.appsmith.server.repositories.OrganizationRepository. If you want this repository to be a Redis repository, consider annotating your entities with one of these annotations: org.springframework.data.redis.core.RedisHash (preferred), or consider extending one of the following types with your repository: org.springframework.data.keyvalue.repository.KeyValueRepository.
appsmith-internal-server_1  |  - Spring Data Redis - Could not safely identify store assignment for repository candidate interface com.appsmith.server.repositories.AssetRepository. If you want this repository to be a Redis repository, consider annotating your entities with one of these annotations: org.springframework.data.redis.core.RedisHash (preferred), or consider extending one of the following types with your repository: org.springframework.data.keyvalue.repository.KeyValueRepository.
appsmith-internal-server_1  |  - Spring Data Redis - Could not safely identify store assignment for repository candidate interface com.appsmith.server.repositories.LayoutRepository. If you want this repository to be a Redis repository, consider annotating your entities with one of these annotations: org.springframework.data.redis.core.RedisHash (preferred), or consider extending one of the following types with your repository: org.springframework.data.keyvalue.repository.KeyValueRepository.
appsmith-internal-server_1  |  - Spring Data Redis - Could not safely identify store assignment for repository candidate interface com.appsmith.server.repositories.NewPageRepository. If you want this repository to be a Redis repository, consider annotating your entities with one of these annotations: org.springframework.data.redis.core.RedisHash (preferred), or consider extending one of the following types with your repository: org.springframework.data.keyvalue.repository.KeyValueRepository.
appsmith-internal-server_1  |  - Spring Data Redis - Could not safely identify store assignment for repository candidate interface com.appsmith.server.repositories.UserRepository. If you want this repository to be a Redis repository, consider annotating your entities with one of these annotations: org.springframework.data.redis.core.RedisHash (preferred), or consider extending one of the following types with your repository: org.springframework.data.keyvalue.repository.KeyValueRepository.
appsmith-internal-server_1  |  - Spring Data Redis - Could not safely identify store assignment for repository candidate interface com.appsmith.server.repositories.ApplicationRepository. If you want this repository to be a Redis repository, consider annotating your entities with one of these annotations: org.springframework.data.redis.core.RedisHash (preferred), or consider extending one of the following types with your repository: org.springframework.data.keyvalue.repository.KeyValueRepository.
appsmith-internal-server_1  |  - Spring Data Redis - Could not safely identify store assignment for repository candidate interface com.appsmith.server.repositories.InviteUserRepository. If you want this repository to be a Redis repository, consider annotating your entities with one of these annotations: org.springframework.data.redis.core.RedisHash (preferred), or consider extending one of the following types with your repository: org.springframework.data.keyvalue.repository.KeyValueRepository.
appsmith-internal-server_1  |  - Spring Data Redis - Could not safely identify store assignment for repository candidate interface com.appsmith.server.repositories.PluginRepository. If you want this repository to be a Redis repository, consider annotating your entities with one of these annotations: org.springframework.data.redis.core.RedisHash (preferred), or consider extending one of the following types with your repository: org.springframework.data.keyvalue.repository.KeyValueRepository.
appsmith-internal-server_1  |  - Spring Data Redis - Could not safely identify store assignment for repository candidate interface com.appsmith.server.repositories.GroupRepository. If you want this repository to be a Redis repository, consider annotating your entities with one of these annotations: org.springframework.data.redis.core.RedisHash (preferred), or consider extending one of the following types with your repository: org.springframework.data.keyvalue.repository.KeyValueRepository.
appsmith-internal-server_1  |  - Spring Data Redis - Could not safely identify store assignment for repository candidate interface com.appsmith.server.repositories.ApiTemplateRepository. If you want this repository to be a Redis repository, consider annotating your entities with one of these annotations: org.springframework.data.redis.core.RedisHash (preferred), or consider extending one of the following types with your repository: org.springframework.data.keyvalue.repository.KeyValueRepository.
appsmith-internal-server_1  |  - Spring Data Redis - Could not safely identify store assignment for repository candidate interface com.appsmith.server.repositories.CollectionRepository. If you want this repository to be a Redis repository, consider annotating your entities with one of these annotations: org.springframework.data.redis.core.RedisHash (preferred), or consider extending one of the following types with your repository: org.springframework.data.keyvalue.repository.KeyValueRepository.
appsmith-internal-server_1  |  - Spring Data Redis - Could not safely identify store assignment for repository candidate interface com.appsmith.server.repositories.ProviderRepository. If you want this repository to be a Redis repository, consider annotating your entities with one of these annotations: org.springframework.data.redis.core.RedisHash (preferred), or consider extending one of the following types with your repository: org.springframework.data.keyvalue.repository.KeyValueRepository.
appsmith-internal-server_1  |  - Finished Spring Data repository scanning in 375ms. Found 0 Redis repository interfaces.
appsmith-internal-server_1  |  - Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.springframework.context.annotation.internalScheduledAnnotationProcessor' defined in org.springframework.scheduling.annotation.SchedulingConfiguration: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.scheduling.annotation.ScheduledAnnotationBeanPostProcessor]: Factory method 'scheduledAnnotationProcessor' threw exception; nested exception is org.springframework.cglib.core.CodeGenerationException: java.lang.ClassFormatError-->Illegal class name "java/lang/Class[]" in class file org/springframework/scheduling/annotation/SchedulingConfiguration$$FastClassBySpringCGLIB$$bd049a56
appsmith-internal-server_1  |  - 
appsmith-internal-server_1  | 
appsmith-internal-server_1  | Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
appsmith-internal-server_1  |  - Application run failed
appsmith-internal-server_1  | org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.springframework.context.annotation.internalScheduledAnnotationProcessor' defined in org.springframework.scheduling.annotation.SchedulingConfiguration: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.scheduling.annotation.ScheduledAnnotationBeanPostProcessor]: Factory method 'scheduledAnnotationProcessor' threw exception; nested exception is org.springframework.cglib.core.CodeGenerationException: java.lang.ClassFormatError-->Illegal class name "java/lang/Class[]" in class file org/springframework/scheduling/annotation/SchedulingConfiguration$$FastClassBySpringCGLIB$$bd049a56
appsmith-internal-server_1  | 	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:656)
appsmith-internal-server_1  | 	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:484)
appsmith-internal-server_1  | 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1338)
appsmith-internal-server_1  | 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1177)
appsmith-internal-server_1  | 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:557)
appsmith-internal-server_1  | 	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:517)
appsmith-internal-server_1  | 	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:323)
appsmith-internal-server_1  | 	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:222)
appsmith-internal-server_1  | 	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:321)
appsmith-internal-server_1  | 	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:207)
appsmith-internal-server_1  | 	at org.springframework.context.support.PostProcessorRegistrationDelegate.registerBeanPostProcessors(PostProcessorRegistrationDelegate.java:228)
appsmith-internal-server_1  | 	at org.springframework.context.support.AbstractApplicationContext.registerBeanPostProcessors(AbstractApplicationContext.java:722)
appsmith-internal-server_1  | 	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:535)
appsmith-internal-server_1  | 	at org.springframework.boot.web.reactive.context.ReactiveWebServerApplicationContext.refresh(ReactiveWebServerApplicationContext.java:66)
appsmith-internal-server_1  | 	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:747)
appsmith-internal-server_1  | 	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:397)
appsmith-internal-server_1  | 	at org.springframework.boot.SpringApplication.run(SpringApplication.java:315)
appsmith-internal-server_1  | 	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1226)
appsmith-internal-server_1  | 	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1215)
appsmith-internal-server_1  | 	at com.appsmith.server.ServerApplication.main(ServerApplication.java:12)
appsmith-internal-server_1  | 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
appsmith-internal-server_1  | 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
appsmith-internal-server_1  | 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
appsmith-internal-server_1  | 	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
appsmith-internal-server_1  | 	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48)
appsmith-internal-server_1  | 	at org.springframework.boot.loader.Launcher.launch(Launcher.java:87)
appsmith-internal-server_1  | 	at org.springframework.boot.loader.Launcher.launch(Launcher.java:51)
appsmith-internal-server_1  | 	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:52)
appsmith-internal-server_1  | Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.scheduling.annotation.ScheduledAnnotationBeanPostProcessor]: Factory method 'scheduledAnnotationProcessor' threw exception; nested exception is org.springframework.cglib.core.CodeGenerationException: java.lang.ClassFormatError-->Illegal class name "java/lang/Class[]" in class file org/springframework/scheduling/annotation/SchedulingConfiguration$$FastClassBySpringCGLIB$$bd049a56
appsmith-internal-server_1  | 	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:185)
appsmith-internal-server_1  | 	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:651)
appsmith-internal-server_1  | 	... 27 common frames omitted
appsmith-internal-server_1  | Caused by: org.springframework.cglib.core.CodeGenerationException: java.lang.ClassFormatError-->Illegal class name "java/lang/Class[]" in class file org/springframework/scheduling/annotation/SchedulingConfiguration$$FastClassBySpringCGLIB$$bd049a56
appsmith-internal-server_1  | 	at org.springframework.cglib.core.ReflectUtils.defineClass(ReflectUtils.java:506)
appsmith-internal-server_1  | 	at org.springframework.cglib.core.AbstractClassGenerator.generate(AbstractClassGenerator.java:363)
appsmith-internal-server_1  | 	at org.springframework.cglib.core.AbstractClassGenerator$ClassLoaderData$3.apply(AbstractClassGenerator.java:110)
appsmith-internal-server_1  | 	at org.springframework.cglib.core.AbstractClassGenerator$ClassLoaderData$3.apply(AbstractClassGenerator.java:108)
appsmith-internal-server_1  | 	at org.springframework.cglib.core.internal.LoadingCache$2.call(LoadingCache.java:54)
appsmith-internal-server_1  | 	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
appsmith-internal-server_1  | 	at org.springframework.cglib.core.internal.LoadingCache.createEntry(LoadingCache.java:61)
appsmith-internal-server_1  | 	at org.springframework.cglib.core.internal.LoadingCache.get(LoadingCache.java:34)
appsmith-internal-server_1  | 	at org.springframework.cglib.core.AbstractClassGenerator$ClassLoaderData.get(AbstractClassGenerator.java:134)
appsmith-internal-server_1  | 	at org.springframework.cglib.core.AbstractClassGenerator.create(AbstractClassGenerator.java:319)
appsmith-internal-server_1  | 	at org.springframework.cglib.reflect.FastClass$Generator.create(FastClass.java:65)
appsmith-internal-server_1  | 	at org.springframework.cglib.proxy.MethodProxy.helper(MethodProxy.java:135)
appsmith-internal-server_1  | 	at org.springframework.cglib.proxy.MethodProxy.init(MethodProxy.java:76)
appsmith-internal-server_1  | 	at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:242)
appsmith-internal-server_1  | 	at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:363)
appsmith-internal-server_1  | 	at org.springframework.scheduling.annotation.SchedulingConfiguration$$EnhancerBySpringCGLIB$$f6ee6cc1.scheduledAnnotationProcessor(<generated>)
appsmith-internal-server_1  | 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
appsmith-internal-server_1  | 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
appsmith-internal-server_1  | 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
appsmith-internal-server_1  | 	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
appsmith-internal-server_1  | 	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:154)
appsmith-internal-server_1  | 	... 28 common frames omitted
appsmith-internal-server_1  | Caused by: java.lang.ClassFormatError: Illegal class name "java/lang/Class[]" in class file org/springframework/scheduling/annotation/SchedulingConfiguration$$FastClassBySpringCGLIB$$bd049a56
appsmith-internal-server_1  | 	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
appsmith-internal-server_1  | 	at java.base/java.lang.System$2.defineClass(Unknown Source)
appsmith-internal-server_1  | 	at java.base/java.lang.invoke.MethodHandles$Lookup.defineClass(Unknown Source)
appsmith-internal-server_1  | 	at java.base/jdk.internal.reflect.GeneratedMethodAccessor19.invoke(Unknown Source)
appsmith-internal-server_1  | 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
appsmith-internal-server_1  | 	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
appsmith-internal-server_1  | 	at org.springframework.cglib.core.ReflectUtils.defineClass(ReflectUtils.java:501)
appsmith-internal-server_1  | 	... 48 common frames omitted
appsmith-internal-server_1 exited with code 1
```

--- a/.github/workflows/github-release.yml
+++ b/.github/workflows/github-release.yml
@@ -217,10 +217,21 @@ jobs:
           name: rts-build-deps
           path: app/rts/node_modules/
 
+      - name: Set up QEMU (needed for docker buildx)
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+
       - name: Build and push client image
         working-directory: app/client
         run: |
-          echo ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }} | docker login -u ${{ secrets.DOCKER_HUB_USERNAME }} --password-stdin
           docker build -t ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-editor:${{needs.prelude.outputs.tag}} .
           docker push ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-editor:${{needs.prelude.outputs.tag}}
 
@@ -230,10 +241,25 @@ jobs:
             docker push ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-editor:latest
           fi
 
+      - name: Build and push fat image
+        run: |
+          tag_args="--tag ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-ce:${{needs.prelude.outputs.tag}}"
+
+          # Only build & tag with latest if the tag doesn't contain beta
+          if [[ ! ${{needs.prelude.outputs.tag}} == *"beta"* ]]; then
+            tag_args="$tag_args --tag ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-ce:latest"
+          fi
+
+          docker buildx build \
+            --platform linux/arm64,linux/amd64 \
+            --push \
+            --build-arg APPSMITH_SEGMENT_CE_KEY=${{ secrets.APPSMITH_SEGMENT_CE_KEY }} \
+            $tag_args \
+            .
+
       - name: Build and push server image
         working-directory: app/server
         run: |
-          echo ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }} | docker login -u ${{ secrets.DOCKER_HUB_USERNAME }} --password-stdin
           docker build --build-arg APPSMITH_SEGMENT_CE_KEY=${{ secrets.APPSMITH_SEGMENT_CE_KEY }} -t ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-server:${{needs.prelude.outputs.tag}} .
           docker push ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-server:${{needs.prelude.outputs.tag}}
 
@@ -247,7 +273,6 @@ jobs:
         working-directory: app/rts
         run: |
           docker build -t ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-rts:${{needs.prelude.outputs.tag}} .
-          echo ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }} | docker login -u ${{ secrets.DOCKER_HUB_USERNAME }} --password-stdin
           docker push ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-rts:${{needs.prelude.outputs.tag}}
 
           # Only build & tag with latest if the tag doesn't contain beta
@@ -255,19 +280,3 @@ jobs:
             docker build -t ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-rts:latest .
             docker push ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-rts:latest
           fi
-
-      - name: Build and push fat image
-        run: |
-          docker build \
-            --build-arg APPSMITH_SEGMENT_CE_KEY=${{ secrets.APPSMITH_SEGMENT_CE_KEY }} \
-            --tag ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-ce:${{needs.prelude.outputs.tag}} \
-            .
-
-          # Only build & tag with latest if the tag doesn't contain beta
-          if [[ ! ${{needs.prelude.outputs.tag}} == *"beta"* ]]; then
-            docker build --tag ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-ce:latest .
-            docker push ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-ce:latest
-          fi
-
-          echo ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }} | docker login -u ${{ secrets.DOCKER_HUB_USERNAME }} --password-stdin
-          docker push --all-tags ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-ce

--- a/.github/workflows/test-build-docker-image.yml
+++ b/.github/workflows/test-build-docker-image.yml
@@ -581,6 +581,33 @@ jobs:
           docker push ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-editor:${GITHUB_SHA}
           docker push ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-editor:nightly
 
+      - name: Build and push release image to Docker Hub
+        if: success() && github.ref == 'refs/heads/release' && (github.event_name == 'push' || github.event_name == 'workflow_dispatch')
+        working-directory: "."
+        run: |
+          tag_args="--tag ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-ce:${{steps.vars.outputs.tag}}"
+
+          docker buildx build \
+            --platform linux/arm64,linux/amd64 \
+            --push \
+            --build-arg APPSMITH_SEGMENT_CE_KEY=${{ secrets.APPSMITH_SEGMENT_CE_KEY }} \
+            $tag_args \
+            .
+
+      - name: Build and push master image to Docker Hub with commit tag
+        if: success() && github.ref == 'refs/heads/master' && (github.event_name == 'push' || github.event_name == 'workflow_dispatch')
+        working-directory: "."
+        run: |
+          tag_args="--tag ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-ce:${GITHUB_SHA}"
+          tag_args="$tag_args --tag ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-ce:nightly"
+
+          docker buildx build \
+            --platform linux/arm64,linux/amd64 \
+            --push \
+            --build-arg APPSMITH_SEGMENT_CE_KEY=${{ secrets.APPSMITH_SEGMENT_CE_KEY }} \
+            $tag_args \
+            .
+
       # Build release Docker image and push to Docker Hub
       - name: Push server release image to Docker Hub
         if: success() && github.ref == 'refs/heads/release'
@@ -616,26 +643,3 @@ jobs:
           docker build -t ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-rts:nightly .
           docker push ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-rts:${GITHUB_SHA}
           docker push ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-rts:nightly
-
-      - name: Build and push release image to Docker Hub
-        if: success() && github.ref == 'refs/heads/release' && (github.event_name == 'push' || github.event_name == 'workflow_dispatch')
-        working-directory: "."
-        run: |
-          docker buildx build \
-            --platform linux/arm64,linux/amd64 \
-            --push \
-            --build-arg APPSMITH_SEGMENT_CE_KEY=${{ secrets.APPSMITH_SEGMENT_CE_KEY }} \
-            --tag ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-ce:${{steps.vars.outputs.tag}} \
-            .
-
-      - name: Build and push master image to Docker Hub with commit tag
-        if: success() && github.ref == 'refs/heads/master' && (github.event_name == 'push' || github.event_name == 'workflow_dispatch')
-        working-directory: "."
-        run: |
-          docker buildx build \
-            --platform linux/arm64,linux/amd64 \
-            --push \
-            --build-arg APPSMITH_SEGMENT_CE_KEY=${{ secrets.APPSMITH_SEGMENT_CE_KEY }} \
-            --tag ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-ce:${GITHUB_SHA} \
-            --tag ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-ce:nightly \
-            .

--- a/.github/workflows/test-build-docker-image.yml
+++ b/.github/workflows/test-build-docker-image.yml
@@ -551,13 +551,24 @@ jobs:
         id: vars
         run: echo ::set-output name=tag::$(echo ${GITHUB_REF:11})
 
+      - name: Set up QEMU (needed for docker buildx)
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+
       # Build release Docker image and push to Docker Hub
       - name: Push client release image to Docker Hub
         if: success() && github.ref == 'refs/heads/release' && (github.event_name == 'push' || github.event_name == 'workflow_dispatch')
         working-directory: app/client
         run: |
           docker build -t ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-editor:${{steps.vars.outputs.tag}} .
-          echo ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }} | docker login -u ${{ secrets.DOCKER_HUB_USERNAME }} --password-stdin
           docker push ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-editor:${{steps.vars.outputs.tag}}
 
       # Build master Docker image and push to Docker Hub
@@ -567,7 +578,6 @@ jobs:
         run: |
           docker build -t ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-editor:${GITHUB_SHA} .
           docker build -t ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-editor:nightly .
-          echo ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }} | docker login -u ${{ secrets.DOCKER_HUB_USERNAME }} --password-stdin
           docker push ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-editor:${GITHUB_SHA}
           docker push ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-editor:nightly
 
@@ -577,7 +587,6 @@ jobs:
         working-directory: app/server
         run: |
           docker build --build-arg APPSMITH_SEGMENT_CE_KEY=${{ secrets.APPSMITH_SEGMENT_CE_KEY }} -t ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-server:${{steps.vars.outputs.tag}} .
-          echo ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }} | docker login -u ${{ secrets.DOCKER_HUB_USERNAME }} --password-stdin
           docker push ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-server:${{steps.vars.outputs.tag}}
 
       # Build master Docker image and push to Docker Hub
@@ -587,7 +596,6 @@ jobs:
         run: |
           docker build --build-arg APPSMITH_SEGMENT_CE_KEY=${{ secrets.APPSMITH_SEGMENT_CE_KEY }} -t ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-server:${GITHUB_SHA} .
           docker build --build-arg APPSMITH_SEGMENT_CE_KEY=${{ secrets.APPSMITH_SEGMENT_CE_KEY }} -t ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-server:nightly .
-          echo ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }} | docker login -u ${{ secrets.DOCKER_HUB_USERNAME }} --password-stdin
           docker push ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-server:${GITHUB_SHA}
           docker push ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-server:nightly
 
@@ -597,7 +605,6 @@ jobs:
         working-directory: app/rts
         run: |
           docker build -t ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-rts:${{steps.vars.outputs.tag}} .
-          echo ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }} | docker login -u ${{ secrets.DOCKER_HUB_USERNAME }} --password-stdin
           docker push ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-rts:${{steps.vars.outputs.tag}}
 
       # Build master Docker image and push to Docker Hub
@@ -607,7 +614,6 @@ jobs:
         run: |
           docker build -t ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-rts:${GITHUB_SHA} .
           docker build -t ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-rts:nightly .
-          echo ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }} | docker login -u ${{ secrets.DOCKER_HUB_USERNAME }} --password-stdin
           docker push ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-rts:${GITHUB_SHA}
           docker push ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-rts:nightly
 
@@ -615,21 +621,21 @@ jobs:
         if: success() && github.ref == 'refs/heads/release' && (github.event_name == 'push' || github.event_name == 'workflow_dispatch')
         working-directory: "."
         run: |
-          docker build \
+          docker buildx build \
+            --platform linux/arm64,linux/amd64 \
+            --push \
             --build-arg APPSMITH_SEGMENT_CE_KEY=${{ secrets.APPSMITH_SEGMENT_CE_KEY }} \
             --tag ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-ce:${{steps.vars.outputs.tag}} \
             .
-          echo ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }} | docker login -u ${{ secrets.DOCKER_HUB_USERNAME }} --password-stdin
-          docker push --all-tags ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-ce
 
       - name: Build and push master image to Docker Hub with commit tag
         if: success() && github.ref == 'refs/heads/master' && (github.event_name == 'push' || github.event_name == 'workflow_dispatch')
         working-directory: "."
         run: |
-          docker build \
+          docker buildx build \
+            --platform linux/arm64,linux/amd64 \
+            --push \
             --build-arg APPSMITH_SEGMENT_CE_KEY=${{ secrets.APPSMITH_SEGMENT_CE_KEY }} \
             --tag ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-ce:${GITHUB_SHA} \
             --tag ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-ce:nightly \
             .
-          echo ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }} | docker login -u ${{ secrets.DOCKER_HUB_USERNAME }} --password-stdin
-          docker push --all-tags ${{ secrets.DOCKER_HUB_ORGANIZATION }}/appsmith-ce

